`deepMerge` seems to ignore merge options on nested props

**Describe the bug**

`deepMerge` seems to ignore merge options on nested props.

**Steps to Reproduce**

```typescript
import { assertEquals } from "std/testing/asserts.ts";
import { deepMerge } from "https://deno.land/std@0.156.0/collections/deep_merge.ts";

Deno.test("deep merge", () => {
  assertEquals(
    deepMerge(
      {
        a: "A1",
        b: ["B11", "B12"],
        c: {
          d: "D1",
          e: ["E11"],
        },
      },
      {
        b: [],
        c: {
          d: "D2",
          e: [],
        },
      },
      {
        arrays: "replace",
      },
    ),
    {
      a: "A1",
      b: [],
      c: {
        d: "D2",
        e: [],
      },
    },
  );
});
```

This test fails with the following output:

```
Check file:///<...>
running 1 test from <...>
deep merge ... FAILED (6ms)

ERRORS 

deep merge => <...>
error: AssertionError: Values are not equal:


    [Diff] Actual / Expected


    {
      a: "A1",
      b: [
      ],
      c: {
        d: "D2",
        e: [
-         "E11",
        ],
      },
    }

  throw new AssertionError(message);
        ^
    at assertEquals (https://deno.land/std@0.154.0/testing/asserts.ts:184:9)
    at file:///<...>

 FAILURES 

deep merge patch => <...>

FAILED | 0 passed | 1 failed (17ms)

error: Test failed
```

**Expected behavior**

The test should pass.

**Environment**

- OS:
- deno version:

```
deno 1.25.3 (release, aarch64-apple-darwin)
v8 10.6.194.5
typescript 4.7.4
```

- std version: 0.156.0

--- a/collections/deep_merge.ts
+++ b/collections/deep_merge.ts
@@ -133,7 +133,7 @@ function mergeObjects(
 ): Readonly<NonNullable<object>> {
   // Recursively merge mergeable objects
   if (isMergeable(left) && isMergeable(right)) {
-    return deepMergeInternal(left, right, seen);
+    return deepMergeInternal(left, right, seen, options);
   }
 
   if (isIterable(left) && isIterable(right)) {

--- a/collections/deep_merge_test.ts
+++ b/collections/deep_merge_test.ts
@@ -299,6 +299,39 @@ Deno.test("deepMerge: sets merge (merge)", () => {
   );
 });
 
+Deno.test("deepMerge: nested replace", () => {
+  assertEquals(
+    deepMerge(
+      {
+        a: "A1",
+        b: ["B11", "B12"],
+        c: {
+          d: "D1",
+          e: ["E11"],
+        },
+      },
+      {
+        b: [],
+        c: {
+          d: "D2",
+          e: [],
+        },
+      },
+      {
+        arrays: "replace",
+      },
+    ),
+    {
+      a: "A1",
+      b: [],
+      c: {
+        d: "D2",
+        e: [],
+      },
+    },
+  );
+});
+
 Deno.test("deepMerge: complex test", () => {
   assertEquals(
     deepMerge({

